Automattic/wp-calypso
# Clean up the Apple Pay user interface

#### Changes proposed in this Pull Request

In preparation for launching Apple Pay, this pull requests fixes a number of issues with the user interface (some based on the guidelines at https://developer.apple.com/design/human-interface-guidelines/apple-pay):
- Replace the Apple Pay icon with the official one from https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/buttons-and-marks#apple-pay-mark (we were using a placeholder before).
- Show the Apple Pay icon on the list of supported payment methods on the Plans page (only for users who have it as an available payment method).
- Replace "Pay with Wallet" text with "Pay with Apple Pay" after Apple Pay has been selected on mobile browsers.
- Add some standard elements to the checkout form that were missing for Apple Pay but that appear for other payment methods (coupon entry, mobile order summary, chat button, etc).
- Stop showing individual line items inside the Apple Pay payment sheet; instead, replace it with just "Subtotal" and "Tax".  It would have been too difficult (and too crowded) to describe each item adequately inside the payment sheet; we would essentially have to reproduce the entire shopping cart in there.  Leaving them off is also recommended in the Apple Pay guidelines (https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/checkout-and-payment#customize-the-payment-sheet), and it should be OK since they're visible in the mobile order summary that was added via the bullet point above.
- Use "Pay WordPress.com" rather than "Pay Total" inside the payment sheet (as recommended by https://developer.apple.com/design/human-interface-guidelines/apple-pay/overview/checkout-and-payment#customize-the-payment-sheet).  This also makes the authorized payment appear sensibly inside the user's Apple Pay wallet.
- When ZIP code or country aren't entered, show an error message when clicking on the Buy with Apple Pay button, rather than silently disabling the button (consistent with Apple Pay user interface guidelines, and also consistent with what we do for other payment methods like PayPal).

#### Testing instructions

Make an Apple Pay purchase and click through all the resulting screens to see the changes.  Screenshots are below.

Before (top) and after (bottom) of the Apple Pay icon:

![apple-pay-icon-old](https://user-images.githubusercontent.com/235183/57749617-cc90b980-76ac-11e9-84cd-326aa8a210f0.png)
![apple-pay-icon-new](https://user-images.githubusercontent.com/235183/57749616-cc90b980-76ac-11e9-9fff-e3b7972f5b3d.png)

Icon appearing underneath the plans list on the Plans page:

![apple-pay-plans-page-new](https://user-images.githubusercontent.com/235183/57749618-cc90b980-76ac-11e9-9e36-227f4ea61ee4.png)

Before (left) and after (right) of the mobile dropdown for selecting a payment method, showing the new icon:

![dropdown-comparison](https://user-images.githubusercontent.com/235183/57749621-cd295000-76ac-11e9-855c-b283703e48b5.png)

Before (left) and after (right) of the dropdown after Apple Pay has been selected:

![apple-pay-selected-comparison](https://user-images.githubusercontent.com/235183/57749619-cc90b980-76ac-11e9-97e1-596a03eb2cb4.png)

Before (left) and after (right) of the items at the bottom of the checkout form:

![buy-with-apple-pay-comparison](https://user-images.githubusercontent.com/235183/57749620-cc90b980-76ac-11e9-96b0-361a9263c31f.png)

Before (left) and after (right) of the payment sheet that appears after clicking through to buy with Apple Pay:

![payment-sheet-comparison](https://user-images.githubusercontent.com/235183/57749740-5e002b80-76ad-11e9-9565-1e580cb47d2a.png)
